src/metabase/native_query_analyzer/replacement.clj

```diff
@@ -44,21 +51,33 @@
       (string? token)
       (recur rest (str query-so-far token) substitutions)
 
-      (params/Param? token)
-      (let [sub (sentinel-variable raw-query)]
-        (recur rest
-               (str query-so-far sub)
```

Comment:
Aside - it looks like we do a lot of string appending, so we should probably refactor to use a `StringBuilder`. If we want to avoid interop we could also use a cons list which we reverse at the end before joining.